nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/ASMain.java

```diff
@@ -17,16 +17,26 @@
 
 package com.sun.enterprise.glassfish.bootstrap;
 
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 /**
  * Tag Main to get the manifest file
  */
 public class ASMain {
 
+    private static final Logger LOG = Logger.getLogger(ASMain.class.getName());
+
     /**
      * Most of the code in this file has been moved to MainHelper
      * and ASMainOSGi
      */
     public static void main(final String[] args) throws Exception {
-        GlassFishMain.main(args);
+        try {
+            GlassFishMain.main(args);
+        } catch (Throwable t) {
+            LOG.log(Level.SEVERE, "Error starting GlassFish", t);
```

Comment:
If GF fails to start, this cannot print anyhing, because GlassFish will not load logging.properties. So in this case t.printStacktrace() would be better.

Reply:
It does actually gets added to the output (not the log file). So you do see what happened if GF doesn't start there. Of course, we could do `t.printStacktrace()` if that's better?